apps/remote_control/lib/lexical/remote_control/code_intelligence/ast/aliases.ex

```diff
@@ -0,0 +1,214 @@
+       ) do
+    # This is for unexpanded aliases, like: alias MyModule.{MyChild, MyOther}
+    unexpanded = unexpanded_alias_mapping(parent_modules, children_ast)
+    {ast, Map.merge(acc, unexpanded)}
+  end
+
+  defp do_collect(
+         {:alias, _,
+          [
+            {{:., _, [{:__MODULE__, position, _}, :{}]}, _, children_ast}
+          ]} = ast,
+         acc,
+         quoted
+       ) do
+    # This is for unexpanded aliases that leading with `__MODULE__`
+    # like: alias __MODULE__.{MyChild, MyOtherChild}
+    {:ok, parent_modules} = fetch_aliases_of_current_module(quoted, position[:line])
+    unexpanded = unexpanded_alias_mapping(parent_modules, children_ast)
+    {ast, Map.merge(acc, unexpanded)}
+  end
+
+  defp do_collect({:defmodule, _, [{:__aliases__, _, aliases} | _]} = ast, acc, quoted) do
+    # This is for implicit aliases, like the `State` module in a GenServer implementation Module
+    # or some nested defmodules
+    {ast, Map.merge(acc, implicit_alias_mapping(quoted, aliases))}
+  end
+
+  defp do_collect(ast, acc, _quoted) do
+    {ast, acc}
+  end
+
+  defp implicit_alias_mapping(quoted, aliases) do
+    path_to_defmodule =
+      Macro.path(
+        quoted,
+        &match?({:defmodule, _, [{:__aliases__, _, ^aliases} | _]}, &1)
+      )
+
+    modules =
+      path_to_defmodule
+      |> Enum.reduce([], fn
+        {:defmodule, _, [{:__aliases__, _, aliases} | _]}, acc ->
+          acc ++ [Module.concat(aliases)]
+
+        _, acc ->
+          acc
+      end)
+
+    modules_length = length(modules)
+
+    modules
+    |> Enum.with_index(fn module, index ->
+      ancestors_to_current_module =
+        modules |> Enum.slice(index, modules_length) |> Enum.reverse() |> Module.concat()
+
+      {module, ancestors_to_current_module}
+    end)
+    |> Map.new()
+  end
+
+  defp cursor_current_module_mapping(quoted) do
+    path_to_cursor = Macro.path(quoted, &match?({:__cursor__, _, _}, &1))
+
+    modules =
+      path_to_cursor
+      |> Enum.reduce([], fn
+        {:defmodule, _, [{:__aliases__, _, aliases} | _]}, acc ->
+          [Module.concat(aliases ++ acc)]
+
+        _, acc ->
+          acc
+      end)
+
+    if modules != [] do
+      [module] = modules
+      %{:__MODULE__ => module}
+    else
+      %{}
+    end
+  end
+
+  defp unexpanded_alias_mapping(parent_modules, children_ast) do
+    for {:__aliases__, _, [child]} <- children_ast, into: %{} do
+      {Module.concat([child]), Module.concat(parent_modules ++ [child])}
+    end
+  end
+
+  defp fetch_aliases_of_current_module(quoted, alias_line) do
+    path =
+      Macro.path(quoted, fn
+        {:defmodule, [line: line], _} when line < alias_line -> true
+        _ -> false
+      end)
+
+    if path do
+      [defs | _] = path
+      {:defmodule, _, [{:__aliases__, _, aliases} | _]} = defs
+      {:ok, aliases}
+    else
+      Logger.info("cannot find the module definition of __MODULE__ in #{alias_line}")
+      :error
+    end
```

Comment:
this might read easier as a case
```elixir

```suggestion
    case path do
      [{:defmodule, _, [{:__aliases__, _, aliases} | _]} | _] -> 
        {:ok, aliases}
    _other ->
      Logger.info("cannot find the module definition of __MODULE__ in #{alias_line}")
      :error
    end
```